x-pack/plugins/infra/public/hooks/use_alerts_count.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useRef } from 'react';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+
+import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common/constants';
+import { estypes } from '@elastic/elasticsearch';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { HttpSetup } from '@kbn/core/public';
+import type { ValidFeatureId } from '@kbn/rule-data-utils';
+
+import { InfraClientCoreStart } from '../types';
+
+interface UseAlertsCountProps {
+  featureIds: ValidFeatureId[];
+  filter?: estypes.QueryDslQueryContainer;
+}
+
+interface FetchAlertsCountParams {
+  featureIds: ValidFeatureId[];
+  filter?: estypes.QueryDslQueryContainer;
+  http: HttpSetup;
+  signal: AbortSignal;
+}
+
+interface AlertsCount {
+  activeAlertCount: number;
+  recoveredAlertCount: number;
+}
+
+export function useAlertsCount({ featureIds, filter }: UseAlertsCountProps) {
+  const { http } = useKibana<InfraClientCoreStart>().services;
+
+  const abortCtrlRef = useRef(new AbortController());
+
+  const [state, refetch] = useAsyncFn(
+    () => {
+      abortCtrlRef.current.abort();
```

Comment:
This happens because the alerts requests depend on both the `hostNodes` and the unified time range, and since `hostNodes` is fetched also on timeRange changes, it causes another sequential call.
I'm canceling the requests if a new one is triggered while still fetching, but I haven't yet a better way to avoid multiple calls since the query should change on both changes to the time and the `hostNodes`